mysensors/mysensors.py

```diff
@@ -552,6 +558,15 @@ def _check_socket(self, sock=None, timeout=None):
             raise OSError
         return available_socks
 
+    def tcp_check(self):
+        if not self._stop_tcp_check.is_set():
+            t = threading.Timer(10.0, self.tcp_check)    #send heartbeat every 10 seconds until socket is disconnected
+            t.setDaemon(True)
+            t.start()
+            _LOGGER.info('Sending I_HEARTBEAT')
+            #self.set_child_internal(0, 255, 2, 0)
+            self.set_child_value(0, 255, 18, 0)
```

Comment:
You should not use this method. Create a Message and use the copy method to set it to internal type etc. Then add it to the FIFO queue with the fill_queue method, while calling the encode method. See https://github.com/afeno/pymysensors/blob/81d11f1f0d750c98819811031fdcbaff08531033/mysensors/mysensors.py#L322-L327.

So then you don't need to change set_child_value method or is_sensor method.

I would prefer if you can use internal message I_VERSION which is supported by mysensors 1.4 and higher. Please test that, if you haven't done it already. The heartbeat messages are new in version 2.0.